### PR TITLE
classifier: fulltext extraction fixes

### PR DESCRIPTION
**Pull request by jalavik**
Some important fixes and cleanup for the legacy classifier/bibclassify integration with regards to the fulltext extraction phase (support for colon in filenames and better reference filter).

### CODE REVIEW

File: invenio/legacy/bibclassify/text_extractor.py [hunk 1]
```diff
@@ -52,7 +50,7 @@
 
 
 def is_pdf(document):
-    """Checks if a document is a PDF file. Returns True if is is."""
+    """Check if a document is a PDF file and returns True if is is."""
```
Comment (jirikuncar) [inline invenio/legacy/bibclassify/text_extractor.py]: `s/returns/return/` for consistency?

File: invenio/modules/classifier/tasks/classification.py [hunk 2]
```diff
@@ -42,11 +42,11 @@ def classify_paper(obj, eng, callback, data,
...
     else:
         result["fast_mode"] = fast_mode
-        if fast_mode:
-            suffix = "fast"
-        else:
-            suffix = "full"
-        name = "classification_{0}".format(suffix)
+        # Check if it is not empty output before adding
+        output = result.get("dict", {}).get("complete_output", {}).values()
+        if not [x for x in output if x]:
```
Comment (jirikuncar) [inline invenio/modules/classifier/tasks/classification.py]: `if not any(output):`?

**Comment by jirikuncar** [comment]
@jalavik `s/BibClassify/classifier/`?

**Comment by jirikuncar** [comment]
- [ ] update copyright years

**Comment by jalavik** [comment]
@jirikuncar Fixes applied and rebased.

File: invenio/modules/classifier/testsuite/taxonomies/microtests.cfg [hunk 3]
```diff
@@ -1,48 +0,0 @@
-[config]
```
Comment (jirikuncar) [inline invenio/modules/classifier/testsuite/taxonomies/microtests.cfg]: Shall we remove also `invenio/legacy/bibclassify/microtests.py`?

Comment (jalavik) [inline invenio/modules/classifier/testsuite/taxonomies/microtests.cfg]: Indeed. Done!

**Pull request merged**